cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -1844,14 +1844,19 @@ func (p *BicepProvider) ensureParameters(
 						paramValue = defValue
 					}
 				}
-				configuredParameters[key] = azure.ArmParameterValue{
+				configuredParameters[key] = azure.ArmParameter{
 					Value: paramValue,
 				}
 				if needForDeployParameter {
 					mustSetParamAsConfig(key, paramValue, p.env.Config, param.Secure())
 					configModified = true
 				}
 				continue
+			} else if v.KeyVaultReference != nil {
```

Comment:
Just a small detail here: I'm wondering how we think about given evaluation priority to `KeyVaultReference` (I think this ends up mirroring the stack_deployments logic):

```golang
if v.KeyVaultReference != nil {
   ...  // set reference
   continue
}

// check if v.Value is set and is assignable to the declared type
paramValue := armParameterFileValue(parameterType, v.Value, param.DefaultValue)
if paramValue != nil {
  ...
  continue
}
```

Reply:
Makes sense, I've updated this